pkg/sink/mysql/config.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	dmysql "github.com/go-sql-driver/mysql"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/contextutil"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/config"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/security"
+	"github.com/pingcap/tiflow/pkg/sink"
+	"go.uber.org/zap"
+)
+
+const (
+	txnModeOptimistic  = "optimistic"
+	txnModePessimistic = "pessimistic"
+
+	// defaultWorkerCount is the default number of workers.
+	defaultWorkerCount = 16
+	// defaultMaxTxnRow is the default max number of rows in a transaction.
+	defaultMaxTxnRow = 256
+	// The upper limit of max worker counts.
+	maxWorkerCount = 1024
+	// The upper limit of max txn rows.
+	maxMaxTxnRow = 2048
+
+	defaultTiDBTxnMode                = txnModeOptimistic
+	defaultBatchReplaceEnabled        = true
+	defaultBatchReplaceSize           = 20
+	defaultReadTimeout                = "2m"
+	defaultWriteTimeout               = "2m"
+	defaultDialTimeout                = "2m"
+	defaultSafeMode                   = true
+	defaultTxnIsolationRC             = "READ-COMMITTED"
+	defaultCharacterSet               = "utf8mb4"
+	defaultDMLMaxRetry         uint64 = 8
```

Comment:
`defaultDMLMaxRetry` isn't a generic option for both `txn.MySQLBackand` and `ddl.MySQLBackend`. If we must seperate `MySQLBackend` into 2 things, I prefer to move `defaultDMLMaxRetry` into `cdc/sinkv2/eventsink/txn/...`.

Reply:
How about rename it to defaultExectueMaxRetry?

Reply:
Seems DDL and DML uses different `maxRetry`.

Reply:
Good catch. Updated.